forms-web-app/src/middleware/fetch-existing-appeal.js

```diff
@@ -9,18 +7,25 @@ const { getExistingAppeal } = require('../lib/appeals-api-wrapper');
  * @param next
  * @returns {Promise<*>}
  */
+const { EMPTY_APPEAL } = require('../lib/appeals-api-wrapper');
+const { getExistingAppeal } = require('../lib/appeals-api-wrapper');
+
 module.exports = async (req, res, next) => {
-  if (!req.session || !req.session.appeal || !req.session.appeal.uuid) {
+  if (!req.session) {
+    return next();
+  }
+
+  if (!req.session.appeal || !req.session.appeal.id) {
+    req.session.appeal = JSON.parse(JSON.stringify(EMPTY_APPEAL));
     return next();
   }
 
   try {
-    req.log.debug({ uuid: req.session.appeal.uuid }, 'Get existing appeal');
-    req.session.appeal = await getExistingAppeal(req.session.appeal.uuid);
+    req.log.debug({ id: req.session.appeal.id }, 'Get existing appeal');
+    req.session.appeal = await getExistingAppeal(req.session.appeal.id);
   } catch (err) {
     req.log.debug({ err }, 'Error retrieving appeal');
-    req.session.appeal = {};
+    req.session.appeal = JSON.parse(JSON.stringify(EMPTY_APPEAL));
```

Comment:
how come you have to parse and stringify?

would it be more helpful if `EMPTY_APPEAL` was already a JSON export? (no idea personally, just throwing it out there)

Reply:
I wanted to have one reference and make a deep copy on it. Sending from a function was painful because of the mock in the test part. But surely there must be a better way to deal with this.